src/FormBuilder/resources/js/form-builder/src/components/canvas/FormSettingsContainer/components/Menu.tsx

```diff
@@ -0,0 +1,63 @@
+import {__} from '@wordpress/i18n';
+import {chevronLeft, chevronRight} from '@wordpress/icons';
+import {Icon} from '@wordpress/components';
+import {useFormSettingsContext} from '@givewp/form-builder/components/canvas/FormSettingsContainer';
+import {
+    navigateBackInMenu,
+    updateMenuState,
+} from '@givewp/form-builder/components/canvas/FormSettingsContainer/formSettingsReducer';
+import classnames from 'classnames';
+
+function MenuItem({item}) {
+    const [state, dispatch] = useFormSettingsContext();
+    const isActive = state.activeMenu === item.path;
+    const hasChildren = item.childRoutes && item.childRoutes.length;
+    const isActiveParent = hasChildren && state.activeMenu.split('/').includes(item.path);
+
+    if (item.showWhen && !item.showWhen()) {
```

Comment:
I could see it being useful to pass the item itself to `showWhen()` to give the function more context. This would allow for multi-use functions.

Reply:
Resolved: 2b3a4b3e0be4a3544305b62a98b68dfc04978209